rust/src/nfs/nfs3_records.rs

```diff
@@ -386,3 +388,703 @@ pub fn parse_nfs3_reply_read(i: &[u8]) -> IResult<&[u8], NfsReplyRead> {
+mod tests {
+    use crate::nfs::nfs3_records::*;
+    use nom7::bytes::streaming::take;
+
+    // helper function to extract nfs3_handle from packets at different offsets
+    fn extract_handle(input: &[u8], offset: usize) -> Nfs3Handle {
+        let i: IResult<&[u8], &[u8]> = take(offset as usize)(input);
+        match i {
+            Ok((r, s)) => {
+                assert_eq!(s.len() as usize, offset);
+                let (_, handle) = parse_nfs3_handle(r).expect("Parsing nfs3_handle failed!");
+                handle
+            }
+            Err(error) => { panic!("extracting nfs3_handle failed {:?}", error); }
+        }
+    }
+
+    #[test]
+    fn test_nfs3_response_create() {
+
+        // packet_bytes
+        let buf: &[u8] = &[
+
+        // Status: NFS3_OK (0)
+            0x00, 0x00, 0x00, 0x00,
+
+        // handle_follows: (1)
+            0x00, 0x00, 0x00, 0x01,
+
+        // [handle]
+        // handle_len: (32)
+            0x00, 0x00, 0x00, 0x20,
+
+        // handle
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x03, 0xe7,
+            0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, 0xa6, 0x54,
+            0x00, 0x00, 0x00, 0x1b, 0x00, 0x0a, 0x00, 0x00,
+            0x00, 0x00, 0xb2, 0x5a, 0x00, 0x00, 0x00, 0x29,
+
+        // unneeded blob
+        /*  0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01,
+            0x00, 0x00, 0x81, 0xa4, 0x00, 0x00, 0x00, 0x01,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x10, 0x10, 0x85,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xa6, 0x54,
+            0x38, 0x47, 0x76, 0x25, 0x22, 0x92, 0x19, 0x01,
+            0x38, 0x47, 0x76, 0x25, 0x22, 0x92, 0x19, 0x01,
+            0x38, 0x47, 0x76, 0x25, 0x22, 0x92, 0x19, 0x01,
+            0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x60, 0x38, 0x47, 0x76, 0x25,
+            0x21, 0xf9, 0x82, 0x82, 0x38, 0x47, 0x76, 0x25,
+            0x21, 0xf9, 0x82, 0x82, 0x00, 0x00, 0x00, 0x01,
+            0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x41, 0xed,
+            0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x60, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0xa3, 0xe7, 0x38, 0x47, 0x76, 0x25,
+            0x21, 0xf9, 0x82, 0x82, 0x38, 0x47, 0x76, 0x25,
+            0x22, 0x92, 0x19, 0x00, 0x38, 0x47, 0x76, 0x25,
+            0x22, 0x92, 0x19, 0x00,
+        */
+        ];
+
+        // asserion: (1) is this a good practice?
+        // manually extract handle from packets each time we parse against it
```

Comment:
I think manual extraction will be required for making the data to match against. We have to generate expected data somehow at some point but you could save `&buf[16..]` here in `value`.

Reply:
@inashivb are you suggesting to manually extract each time we need to match against  `Nfs3Handle` struct?

Reply:
@ghostinthehive if there's another and/or better way to do this, I'll be happy to learn about that. :) But for now, yes, it seems like the way to me.